### PR TITLE
Standardize the nullability of arguments to raw key data constructors.

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This standardizes the handling of nullability for the `LogicalKeyboardKey.keyLabel` and `RawKeyEventData.keyLabel` accessors so that they are non-nullable, but can be empty.

Before this change, the keyLabel could be either null or an empty string to indicate that there wasn't a label, which makes it harder to test for, since both need to be checked for. Since an empty string is sufficient, there is no need for it to be nullable.

Also, in `raw_keyboard.dart`, the web and Windows implementations wouldn't accept null values for parameters in the Map coming from the message, but tests were supplying null for some of them.  This makes web and Windows creation of events match the other platforms, and makes the migration of tests to non-nullability easier.

## Tests

- Updated tests for raw keyboard to look for empty `keyLabel`s instead of null.

## Breaking Change

- [X] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

File: packages/flutter_test/lib/src/event_simulation.dart [hunk 1]
```diff
@@ -209,7 +209,7 @@ class KeyEventSimulator {
       case 'windows':
         result['keyCode'] = keyCode;
         result['scanCode'] = scanCode;
-        result['characterCodePoint'] = key.keyLabel?.codeUnitAt(0) ?? 0;
+        result['characterCodePoint'] = key.keyLabel.isEmpty ? 0 : key.keyLabel?.codeUnitAt(0);
```
Comment (dkwingsmt) [inline packages/flutter_test/lib/src/event_simulation.dart:212]: Why do we still need `?` after `key.keyLabel` here?

Comment (dkwingsmt) [reply packages/flutter_test/lib/src/event_simulation.dart:212]: Ok I see you've explained it in the PR description. But how come it doesn't trigger lint error?

**Review by dkwingsmt — APPROVED** [review]
LGTM

File: packages/flutter_test/lib/src/event_simulation.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter_test/lib/src/event_simulation.dart:212]: You're right, I don't need the `?`.  Not sure why there's no linter error.

**Pull request merged**